### PR TITLE
testing/firefox: upgrade to 67.0.4

### PR DESCRIPTION
**Pull request by timbru31**
This updates Firefox to 67.0.4

### CODE REVIEW

**Comment by Cogitri** [comment]
I realize this is still a WIP; but please don't merge this before #8628, the build can SIGSEV randomly with LLVM7.
Also, please enable rust-simd once that PR is merged since that significantly speeds up some ops in FF. Thanks!

**Comment by mi2iec** [comment]
> Also, please enable rust-simd once that PR is merged since that significantly speeds up some ops in FF. Thanks!
 
Firefox 67.0.2 won't build if the `--enable-rust-simd` flag is enabled:

    ERROR: --enable-rust-simd does not work with Rust 1.33 or later. See https://bugzilla.mozilla.org/show_bug.cgi?id=1521249

**Comment by Cogitri** [comment]
Oh, oops, read that as Rust 1.33 or earlier somehow :)

I guess we'll have to wait until FF 68.x for this then.

**Comment by timbru31** [comment]
Strange. PR fails because of llvm-objdump:

```
checking for llvm-objdump... not found
--
588 | DEBUG: llvm_objdump: Trying llvm-objdump
589 | ERROR: Cannot find llvm-objdump
```

Buuut, it should be available?!

**Comment by Cogitri** [comment]
It works for you locally?

**Comment by mi2iec** [comment]
> Strange. PR fails because of llvm-objdump:
> 
> ```
> checking for llvm-objdump... not found
> --
> 588 | DEBUG: llvm_objdump: Trying llvm-objdump
> 589 | ERROR: Cannot find llvm-objdump
> ```
> 
> Buuut, it should be available?! https://pkgs.alpinelinux.org/contents?file=llvm-objdump&path=&name=llvm*&branch=edge&repo=community&arch=x86_64

In makedepends llvm5-dev should be changed to llvm5 (or llvm8) as llvm-objdump is not in the -dev packages.

**Comment by timbru31** [comment]
```
checking for NSPR - version >= 4.21... no
--
691 | configure: error: you do not have NSPR installed or your version is older than 4.21.
```

We need a nspr package update first: https://pkgs.alpinelinux.org/package/edge/main/x86_64/nspr

File: testing/firefox/APKBUILD [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # Contributor:
 # Maintainer: Natanael Copa <ncopa@alpinelinux.org>
 pkgname=firefox
-pkgver=66.0.5
+pkgver=67.0.3
```
Comment (J0WI) [inline testing/firefox/APKBUILD]:
```suggestion
pkgver=67.0.4
```

File: testing/firefox/APKBUILD [hunk 2]
```diff
@@ -37,7 +37,7 @@ makedepends="
 	libvorbis-dev
 	libxt-dev
 	libxcomposite-dev
-	llvm5-dev
+	llvm8
```
Comment (Cogitri) [inline testing/firefox/APKBUILD]: I think we still need -dev for llvm-config

**Comment by Cogitri** [comment]
Please change the license to `GPL-3.0-only AND MPL-2.0 AND  LGPL-2.1-only AND LGPL-3.0-only` (see toolkit/content/license.html`.

File: testing/firefox/APKBUILD (hunk 2 above)
Comment (timbru31) [reply testing/firefox/APKBUILD]: Passed without it :)

**Comment by Cogitri** [comment]
Alright, could you make this a non-draft? I think this should be OK and having the security fixes in the repos would be nice.

**Comment by liqa** [comment]
community/firefox-esr has cleanups that should be applied here.
https://github.com/alpinelinux/aports/blob/master/community/firefox-esr/APKBUILD
Both packages should be synced with differences only where really necessary.

**Comment by liqa** [comment]
You can delete _pkgver=$pkgver and _xulver=$pkgver.
You can delete empty depends="" and empty install=""
You can drop the custom prepare (the bug mentioned is resolved upstream).
You can delete the "$CARCH" = "x86" part as firefox is not built for that arch.

There should be more cleanups as firefox is for a long time in the repos. I suspect a few other commands in the package are cruft from the past.

**Comment by timbru31** [comment]
Thanks @liqa, I will apply those cleanup advises in this PR!

**Comment by Cogitri** [comment]
Mind taking a look at the CI failure?

**Comment by timbru31** [comment]
Done!

**Comment by Cogitri** [comment]
Nice, mind squashing the commits together then? Then it looks good to me.

**Pull request closed without merging**